src/Lykke.Service.BlockchainWallets/Modules/ServiceModule.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Autofac;
+using Common.Log;
+using Lykke.Service.BlockchainWallets.Core.Services;
+using Lykke.Service.BlockchainWallets.Core.Settings.BlockchainIntegrationSettings;
+using Lykke.Service.BlockchainWallets.Services;
+
+
+namespace Lykke.Service.BlockchainWallets.Modules
+{
+    public class ServiceModule : Module
+    {
+        private readonly ILog                           _log;
+        private readonly BlockchainsIntegrationSettings _blockchainsIntegrationSettings;
+
+
+        public ServiceModule(
+            BlockchainsIntegrationSettings blockchainsIntegrationSettings,
+            ILog log)
+        {
+            _blockchainsIntegrationSettings = blockchainsIntegrationSettings;
+            _log                            = log;
+        }
+
+        protected override void Load(ContainerBuilder builder)
+        {
+            builder
+                .Register(ctx => _blockchainsIntegrationSettings)
+                .AsSelf()
+                .SingleInstance();
+
+            builder
+                .RegisterInstance(_log)
+                .As<ILog>()
+                .SingleInstance();
+            
+            builder
+                .RegisterType<BlockchainIntegrationService>()
+                .As<IBlockchainIntegrationService>()
+                .SingleInstance();
```

Comment:
Why single instance?

Reply:
We use immutables, service is thread safe, so we not need more then one instance